lib/ansible/modules/cloud/amazon/ecs_service_facts.py

```diff
@@ -169,9 +170,18 @@ def list_services(self, cluster):
         fn_args = dict()
         if cluster and cluster is not None:
             fn_args['cluster'] = cluster
-        response = self.ecs.list_services(**fn_args)
-        relevant_response = dict(services = response['serviceArns'])
-        return relevant_response
+
+        # boto3 gives paginated reponse of 10 entries each
```

Comment:
Please use the [ECS ListServices paginator](http://boto3.readthedocs.io/en/latest/reference/services/ecs.html#ECS.Paginator.ListServices) for this. It might also be worth using `AWSRetry` in case of API throttling.

We should really document this, but in the absence of documentation, see e.g 
https://github.com/ansible/ansible/blob/b5eca624d0029bc492a99525ac1ef0c2e469533e/lib/ansible/modules/cloud/amazon/aws_waf_facts.py#L188-L191

for the use of pagination with backoff.

Reply:
oh, didn't know about this, I'll have a look into it!

Reply:
Documentation now exists (thanks for reminding me that I should create it :) )

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination